### PR TITLE
Purge urls filter

### PR DESCRIPTION
**Pull request by diegofs-1989**
Added a regex to the purge method.

### CODE REVIEW

**Review by cssjoe — CHANGES_REQUESTED** [review]
@diegofs-1989 There are some changes that revert recent updates (2 other files).  It appears that the new code change is only in `Cdnfsd_TransparentCDN_Engine.php`.  Please ensure that you checkout the latest `master` branch to only includes the new URL filter code.

**Review by diegofs-1989 — COMMENTED** [review]
Deleted repeated lines and kept the regex.

File: Cdnfsd_TransparentCDN_Engine.php [hunk 1]
```diff
@@ -81,10 +81,13 @@ public function purge( $urls ) {
 		$this->_get_token();
 
 		$invalidation_urls = array();
-
+		//Included a regex filter because some of our clients reported receiving urls as "True" or "False"
+		$url_pattern = "/https?:\/\/(www\.)?[-a-zA-Z0-9]{1,256}\.?[a-zA-Z0-9]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/";
```
Comment (cssjoe) [inline Cdnfsd_TransparentCDN_Engine.php:85]: The PCRE regex has an issue (Unknown modifier '/').  Review these tests:

```
$ php -r '$url = "https://www.exmaple.com/test-page?queryarg=123"; $url_pattern = "/https?:\/\/(www\.)?[-a-zA-Z0-9]{1,256}\.?[a-zA-Z0-9]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/"; var_dump( preg_match( $url_pattern, $url ) );'
Warning: preg_match(): Unknown modifier '/' in Command line code on line 1
bool(false)
```

When removing the two slashes in the pattern:
```
$ php -r '$url = "https://www.exmaple.com/test-page?queryarg=123"; $url_pattern = "/https?:\/\/(www\.)?[-a-zA-Z0-9]{1,256}\.?[a-zA-Z0-9]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&=]*)/"; var_dump( preg_match( $url_pattern, $url ) );'
int(1)
```

If you intended to have the slashes in the character class, then you need only one and need to either escape it or change your regex delimiter.

File: Cdnfsd_Core.php [hunk 2]
```diff
@@ -19,11 +19,10 @@ function get_engine() {
 			case 'cloudflare':
 				$engine_object = null;   // extension handles everything
 				break;
-
-				case 'transparentcdn':
```
Comment (cssjoe) [inline Cdnfsd_Core.php:23]: The changes in this file are not needed.  They may have been from an old copy.

**Review by cssjoe — CHANGES_REQUESTED** [review]
Please see the comments made on the files changed.  Also, if you only need to validate the URL, you can use `if ( filter_var( $url, FILTER_VALIDATE_URL ) ) {` as the condition.

File: Cdnfsd_TransparentCDN_Engine.php [hunk 3]
```diff
@@ -81,10 +81,12 @@ public function purge( $urls ) {
 		$this->_get_token();
 
 		$invalidation_urls = array();
-
+		//Included a regex filter because some of our clients reported receiving urls as "True" or "False"
 		foreach ( $urls as $url ) {
 			//Oh array_map+lambdas, how I miss u...
-			$invalidation_urls[] = $url;
+			if ( filter_var( $url, FILTER_VALIDATE_URL ) ){
```
Comment (cssjoe) [inline Cdnfsd_TransparentCDN_Engine.php:87]: Minor: Missing space before opening brace.

File: Cdnfsd_Core.php (hunk 2 above)
Comment (cssjoe) [reply Cdnfsd_Core.php:23]: This file still has formatting changes.  Please revert to the current copy in master: https://raw.githubusercontent.com/W3EDGE/w3-total-cache/master/Cdnfsd_Core.php

**Review by cssjoe — CHANGES_REQUESTED** [review]
The file `Cdnfsd_Core.php` still needs to be reverted to the current version from our master branch.  There is also a minor code standard issue; a missing space before an opening brace.  See the other comments.  Thanks.

**Review by cssjoe — APPROVED** [review]
@diegofs-1989 Thank you for your contribution.  This update will be included in the upcoming release 2.1.2 in the next couple of days.

**Pull request merged**